### ISSUE TITLE
How to set docker registry for addonComponents

### ISSUE THREAD

**Issue by snowneji** [istio/istio#29283]
**Bug description**

I'm trying to install Istio on AKS under  our company environment, which means we need to refer the internal proxy of docker registries, and I'm following this link: [https://docs.microsoft.com/en-us/azure/aks/servicemesh-istio-install?pivots=client-operating-system-macos](https://docs.microsoft.com/en-us/azure/aks/servicemesh-istio-install?pivots=client-operating-system-macos) and [https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#IstioComponentSetSpec ](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#IstioComponentSetSpec )

While constructing the IstioOperatorSpec, I'm able to get the basic to work with the `hub` argument, but I'm not able to point `addonComponents` such as grafana to the correct Docker Registry:

```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: istio-control-plane
spec:
  hub: INTERNAL_DOCKER_HUB
  profile: default
  # Enable the addons that we will want to use
  addonComponents:
    grafana:
      enabled: true
      hub: INTERNAL_DOCKER_HUB

```

[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Comment by morvencao** [comment]
Which istio version are you using?
Istio addon Integrations was reworked from 1.6.
Istio stopped shipping installation artifacts related to addons. However, Istio will guarantee version compatibility where necessary. It is the user’s responsibility to install these components by using the official Integrations [documentation](https://istio.io/latest/docs/ops/integrations/) and artifacts provided by the respective projects.

**Comment by snowneji** [comment]
Thanks for the response. I was trying to install Kubeflow and the version of Istio was `1.6.13`

The problem I have is that our k8s cluster is under company VPN so we cannot reach directly to public docker hub to fetch those images. When I specify the `hub` argument to `hub: COMPANY_PROXY_OF_HUB/istio`, it worked to install basic Istio to k8s.  But when I need to add addonComponents such as grafana, I saw the installation failed because the pod was trying to pull image from`PUBLIC_HUB/grafana`, which is different from what I specified above, so I'm wondering how can I set the docker hub just for grafana?
